### PR TITLE
Jetpack banner scroll animation

### PR DESCRIPTION
**Pull request by irfano**
This adds scroll animation to the Jetpack banner. 
- At first, the banner is visible. 
- When scrolled down, the banner moves down and disappears.
- When scrolled to the top position again, the banner moves up.
- If the tab changes, the banner behaves according to the current state of the selected tab.

https://user-images.githubusercontent.com/2471769/179851263-0df1dc61-ae62-4376-95a6-e1f5fc49adbf.mp4

To test:
Setup:
1. Launch the WordPress app
2. Go to App Settings (Tap on Avatar at the top right-hand corner on My Site -> Find App Settings)
3. Select Debug Settings
4. Find JetpackPoweredFeatureConfig under Features in development as shown on this PR https://github.com/wordpress-mobile/WordPress-Android/pull/16908
5. Re-launch the app (scroll down if necessary)

Test:
1. Navigate to Reader.
2. Ensure the jetpack banner is visible.
3. Scroll down. Ensure the banner disappeared with an animation.
4. Scroll to the top again. Ensure the banner appears with an animation.
5. Change the tab and ensure the banner is hidden if the list is scrolled, or visible if the scroll is at the top.
6. Naviaget Notifications screen and repeat 2-5.

## Regression Notes
1. Potential unintended areas of impact
Nothing.

6. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

7. What automated tests I added (or what prevented me from doing so)
This is an animation that we don't need a UI test.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ravishanker — APPROVED** [review]
Nice work 🚀 

Feel free to merge it after build is successful

**Review by ovitrif — APPROVED** [review]
Good work @irfano , you have my approval to merge this too once the CI checks are 🟢!

It works as intended and it's really nice to see it animating 💯

**Pull request merged**
